### ISSUE TITLE
Data not prepared correctly?

### ISSUE THREAD

**Issue by culurciello** [karandwivedi42/imagenet-multiGPU.torchnet#3]
elab@GPU8 ~/euge/imagenet-multiGPU.torchnet [master*]$ th utils/datagen.lua -s /media/SuperSSD/ILSVRC2012/ -n ImageNet

elab@GPU8 ~/euge/imagenet-multiGPU.torchnet [master*]$ th train.lua -d /media/SuperSSD/ILSVRC2012 -c ImageNet.t7 -m models/alexnet.lua              
{
  nesterov : true
  nGPU : 1
  factor : 99
  cache : "ImageNet.t7"
  batchSize : 128
  learningRate : 0.01
  nThread : 4
  manualSeed : 1
  data : "/media/SuperSSD/ILSVRC2012"
  weightDecay : 0.0005
  epoch_step : 5
  maxepoch : 100
  save : "logs"
  dampening : 0
  momentum : 0.9
  model : "models/alexnet.lua"
}
nn.Sequential {
  [input -> (1) -> (2) -> (3) -> (4) -> (5) -> (6) -> (7) -> (8) -> (9) -> (10) -> (11) -> (12) -> (13) -> (14) -> (15) -> (16) -> (17) -> (18) -> (19) -> (20) -> (21) -> (22) -> (23) -> (24) -> (25) -> (26) -> (27) -> (28) -> (29) -> output]
  (1): cudnn.SpatialConvolution(3 -> 64, 11x11, 4,4, 2,2)
  (2): cudnn.SpatialBatchNormalization
  (3): cudnn.ReLU
  (4): cudnn.SpatialMaxPooling(3x3, 2,2)
  (5): cudnn.SpatialConvolution(64 -> 192, 5x5, 1,1, 2,2)
  (6): cudnn.SpatialBatchNormalization
  (7): cudnn.ReLU
  (8): cudnn.SpatialMaxPooling(3x3, 2,2)
  (9): cudnn.SpatialConvolution(192 -> 384, 3x3, 1,1, 1,1)
  (10): cudnn.SpatialBatchNormalization
  (11): cudnn.ReLU
  (12): cudnn.SpatialConvolution(384 -> 256, 3x3, 1,1, 1,1)
  (13): cudnn.SpatialBatchNormalization
  (14): cudnn.ReLU
  (15): cudnn.SpatialConvolution(256 -> 256, 3x3, 1,1, 1,1)
  (16): cudnn.SpatialBatchNormalization
  (17): cudnn.ReLU
  (18): cudnn.SpatialMaxPooling(3x3, 2,2)
  (19): nn.View(9216)
  (20): nn.Dropout(0.500000)
  (21): nn.Linear(9216 -> 4096)
  (22): cudnn.BatchNormalization
  (23): cudnn.ReLU
  (24): nn.Dropout(0.500000)
  (25): nn.Linear(4096 -> 4096)
  (26): cudnn.BatchNormalization
  (27): cudnn.ReLU
  (28): nn.Linear(4096 -> 1000)
  (29): cudnn.LogSoftMax
}
/home/elab/torch/install/bin/luajit: train.lua:182: attempt to call method 'size' (a nil value)
stack traceback:
        train.lua:182: in function 'hooks'
        ...ch/install/share/lua/5.1/torchnet/engine/optimengine.lua:100: in function 'train'
        train.lua:237: in main chunk
        [C]: in function 'dofile'
        ...elab/torch/install/lib/luarocks/rocks/trepl/scm-1/bin/th:145: in main chunk
        [C]: at 0x004065d0

**Comment by culurciello** [comment]
Uhm, for my case, in https://github.com/karandwivedi42/imagenet-multiGPU.torchnet/blob/master/train.lua#L182 state.sample.input is a table, not a tensor, so size() and copy() do not work

**Comment by karandwivedi42** [comment]
In the :transform{} I have commented out everything, please uncomment that ( At least the resizing and cropping commands). That way all images will have the same dimensions and then BatchDataset will group them into a tensor automatically rather than table.

**Comment by culurciello** [comment]
ok thanks a lot, that should be in instructions then!
well done!!!
